src/components/Contexts/Variables/Modals/Instrument.tsx

```diff
@@ -32,8 +33,8 @@ export function Instrument() {
       setPort(0);
       getPorts({
         variables: { name: name },
-        onCompleted: (data: { distinctPorts: { issPort: number }[] }) => {
-          setPortOptions(data.distinctPorts.map((e) => e.issPort));
+        onCompleted: (data) => {
+          setPortOptions(data.distinctPorts?.map((e) => e?.issPort).filter(isNotNullish) ?? []);
```

Comment:
There are a lot of changes like this. I think the schema might be too lenient on null-values. If it's stricter we can remove a lot of this (and linting will tell us it's unnecessary)

Reply:
Totally agree

Reply:
I've improved this in #12 but the PR in navigate-configs should be running first